### PR TITLE
Fix VMSS cache content

### LINKED ISSUE #89025: Azure: Many invalid VMSS update calls due to incorrect VMSS cache
**What happened**:
Controller manager makes many invalid VMSS update calls.

**What you expected to happen**:
Controller manager makes correct VMSS update calls.

**How to reproduce it (as minimally and precisely as possible)**:
Use AKS-Engine to deploy a cluster on Azure.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
1.16.6
- Cloud provider or hardware configuration:
Azure
- OS (e.g: `cat /etc/os-release`):
- Kernel (e.g. `uname -a`):
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by ArchangelSDY**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:
Using reference to a loop iterator variable is a common mistake because it
takes different values in each loop iteration.

**Which issue(s) this PR fixes**:

Fixes #89025 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Fix invalid VMSS updates due to incorrect cache
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs

```

### CODE REVIEW

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_cache.go [hunk 1]
```diff
@@ -73,7 +73,8 @@ func (ss *scaleSet) newVMSSCache() (*azcache.TimedCache, error) {
 				return nil, rerr.Error()
 			}
 
-			for _, scaleSet := range allScaleSets {
+			for i := range allScaleSets {
+				scaleSet := allScaleSets[i]
```
Comment (andyzhangx) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_cache.go]: in this loop, `scaleSet ` does not change, it's read-only

Comment (andyzhangx) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_cache.go]: looks like you mean this code:
```
vmss:       &scaleSet,
```

Comment (andyzhangx) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_cache.go]: you are correct:
https://github.com/golang/go/wiki/CommonMistakes#using-reference-to-loop-iterator-variable

**Review by andyzhangx — APPROVED** [review]
/lgtm
/approve
we need to check other places and also cherry-pick to old releases.

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_cache.go (hunk 1 above)
Comment (ArchangelSDY) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_cache.go]: yes, you get it.

**Comment by andyzhangx** [comment]
/hold
hold a while for all tests running pass

**Comment by andyzhangx** [comment]
@ArchangelSDY could you add release-note, create an issue and link that issue to this PR? we should cherry-pick this PR to old releases. thanks.

**Comment by ArchangelSDY** [comment]
Sure. Updated.

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_cache.go (hunk 1 above)
Comment (feiskyer) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_vmss_cache.go]: Good catch, thanks for the fix

**Review by feiskyer — APPROVED** [review]

**Comment by feiskyer** [comment]
Add to 1.18 milestone since this fixes a critical bug

/milestone v1.18
/priority critical-urgent

**Pull request merged**